plugins/orchestrator/src/components/WorkflowDefinitionViewerPage/WorkflowDefinitionViewerPage.tsx

```diff
@@ -61,13 +67,16 @@ export const WorkflowDefinitionViewerPage = () => {
             {loading ? (
               <Skeleton variant="text" width="5rem" />
             ) : (
-              <Button
-                variant="contained"
-                color="primary"
-                onClick={handleExecute}
-              >
-                Run
-              </Button>
+              !loadingPermission && (
+                <Button
+                  variant="contained"
+                  color="primary"
+                  onClick={handleExecute}
+                  disabled={!canRun}
```

Comment:
while it's loading it's pretty easy to add a spinner to the button 
see the SubmitButton used in the Execute page for reference
https://github.com/janus-idp/backstage-plugins/blob/main/plugins/orchestrator/src/components/SubmitButton.tsx#L31